base_report_docx/models/ir_actions.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import fields, models
+
+
+class IrActionsReportXml(models.Model):
+    _inherit = 'ir.actions.report.xml'
+
+    report_type = fields.Selection(
+        [
+            ('qweb-pdf', 'PDF'),
+            ('qweb-html', 'HTML'),
+            ('controller', 'Controller'),
+            ('pdf', 'RML pdf (deprecated)'),
+            ('sxw', 'RML sxw (deprecated)'),
+            ('webkit', 'Webkit (deprecated)'),
+            ('docx', 'Docx'),
+        ],
+        'Report Type', required=True,
+        help="""
+            HTML will open the report directly in your browser,
+            PDF will use wkhtmltopdf to render the HTML into a PDF file
+            and let you download it,
+            Controller allows you to define the url of a custom controller
+            outputting any kind of report.
+            Docx allows you to upload Docx word template in Odoo
+            and get it rendered in PDF or docx.
+            """
+    )
+
+    template_file = fields.Many2one(
+        comodel_name='ir.attachment', string='Template File')
+
+    watermark_string = fields.Char(string='Wartermark String')
```

Comment:
It provides  a second solution for the user. User can simply add the string as the watermark instead of the prepare the template.